festim/hydrogen_transport_problem.py

```diff
@@ -158,6 +160,37 @@ def define_markers_and_measures(self):
             "dx", domain=self.mesh.mesh, subdomain_data=self.volume_meshtags
         )
 
+    def check_borders(self, mesh):
```

Comment:
should this be a method of `Mesh1D` ?

we would pass all the subdomains as argument.

Something like

```python
# inside Mesh1D
def check_borders(self, list_of_subdomains1D):
    ....

# inside Problem

def define_markers_and_measures(self):
    ....
    if isinstance(self.mesh, F.Mesh1D):
        self.mesh.check_borders(self.subdomains)
```